### PR TITLE
Add api conection for water-fall-chart

### PR DESCRIPTION
**Pull request by alejandrocabriales**
## Ticket
https://trello.com/c/qmqD8fNL/311-rc-1-reserves-chart

## Description
Add the connection waterfall chart with the API

## What solved
- [X] :Should display the data coming from the API.
- [X] :Should display the proper data for each type of finance page selected.
- [X] : Should display the proper data for each period selected.
- [X] : Should display the label in the direction of the bar (green on the top and red on the bottom)
- [X] :Should apply for green bars: green font and for red bars: red font. Add the positive or negative sign depending on the case

### CODE REVIEW

File: src/stories/containers/Finances/components/SectionPages/ReservesWaterFallChartSection/useReservesWaterFallChart.tsx [hunk 1]
```diff
@@ -28,9 +47,11 @@ export const useReservesWaterFallChart = (levelPath: string | null, budgets: Bud
...
   }, [selectAll]);
 
-  const { isLight } = useThemeContext();
-  const isMobile = useMediaQuery(lightTheme.breakpoints.down('tablet_768'));
-  const isTable = useMediaQuery(lightTheme.breakpoints.between('tablet_768', 'desktop_1024'));
+  // fetch actual data from the API
+  const { data: analytics, error } = useSWRImmutable([selectedGranularity, year, codePath, levelOfDetail], async () =>
+    fetchAnalytics(selectedGranularity, year, codePath, levelOfDetail)
+  );
+  const isLoading = !analytics && !error;
```
Comment (YasielCabrera) [inline src/stories/containers/Finances/components/SectionPages/ReservesWaterFallChartSection/useReservesWaterFallChart.tsx:54]: with the latest update of `SWR` we can get the `isLoading` value directly from the `useSWRImmutable` response

File: src/stories/containers/Finances/components/SectionPages/ReservesWaterFallChartSection/useReservesWaterFallChart.tsx [hunk 2]
```diff
@@ -1,22 +1,41 @@
...
+  year: string
+) => {
+  const selectAll = useMemo(() => budgets.map((budget) => budget.codePath), [budgets]);
+  const { isLight } = useThemeContext();
+  const [selectedGranularity, setSelectedGranularity] = useState<AnalyticGranularity>('monthly');
+  const isMobile = useMediaQuery(lightTheme.breakpoints.down('tablet_768'));
+  const isTable = useMediaQuery(lightTheme.breakpoints.between('tablet_768', 'desktop_1024'));
+  const router = useRouter();
+  const urlPath = Array.isArray(router.query.path) ? router.query.path.join('/') : router.query.path;
+  const codePath = urlPath ? `atlas/${urlPath}` : 'atlas';
```
Comment (YasielCabrera) [inline src/stories/containers/Finances/components/SectionPages/ReservesWaterFallChartSection/useReservesWaterFallChart.tsx:35]: is not `codePath` and the prop `levelPath` the same thing?

**Review by YasielCabrera — CHANGES_REQUESTED** [review]

File: src/stories/containers/Finances/components/SectionPages/ReservesWaterFallChartSection/useReservesWaterFallChart.tsx (hunk 1 above)
Comment (alejandrocabriales) [reply src/stories/containers/Finances/components/SectionPages/ReservesWaterFallChartSection/useReservesWaterFallChart.tsx:54]: you are right

File: src/stories/containers/Finances/components/SectionPages/ReservesWaterFallChartSection/useReservesWaterFallChart.tsx (hunk 2 above)
Comment (alejandrocabriales) [reply src/stories/containers/Finances/components/SectionPages/ReservesWaterFallChartSection/useReservesWaterFallChart.tsx]: you are right

**Pull request merged**